05week/checkers.js

```diff
@@ -53,16 +64,120 @@ class Board {
   }
 
   // Your code here
+  // Here Should take positions of white pieces, setinng as arrays
+  createCheckers(){
+    // [row, column]
+    const whitePosition = [
+      [0, 1],
+      [0, 3],
+      [0, 5],
+      [0, 7],
+      [1, 0],
+      [1, 2],
+      [1, 4],
+      [1, 6],
+      [2, 1],
+      [2, 3],
+      [2, 5],
+      [2, 7]
+    ]
+    //the loop should set W on board
+    for (let i = 0; i < 12; i++) {
+      let whiteRow = whitePosition[i][0];
+      let whiteColumn = whitePosition[i][1];
+      let whiteChecker = new Checker('white');
+      this.checkers.push(whiteChecker); 
+      this.grid[whiteRow][whiteColumn] = whiteChecker;
+    }
+      // this will target the black pieces on board
+    const blackPosition = [
+      [5, 0],
+      [5, 2],
+      [5, 4],
+      [5, 6],
+      [6, 1],
+      [6, 3],
+      [6, 5],
+      [6, 7],
+      [7, 0],
+      [7, 2],
+      [7, 4],
+      [7, 6]
+    ]
+    // the loop will set B pieces on the board
+    for (let i = 0; i < 12; i++) {
+      let blackRow = blackPosition[i][0];
+      let blackColumn = blackPosition[i][1];
+      let blackChecker = new Checker('black');
+      this.checkers.push(blackChecker);
+      this.grid[blackRow][blackColumn] = blackChecker;
+    }
+  }
 }
+  
 
+//the class Game will hold the move piece function and create the rules foe moving on the board
 class Game {
   constructor() {
     this.board = new Board;
   }
   start() {
     this.board.createGrid();
+    this.board.createCheckers();
+  }
+  moveChecker(source, destination){
```

Comment:
Try breaking this function down into smaller pieces.  For example, write a method that handles jumps, one that handles a single move.  Currently, the ability to jump isn't 100% and the logic currently limits the ability to move across the board.